core/src/main/java/jenkins/management/Badge.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2022, Markus Winter
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.management;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.diagnosis.OldDataMonitor;
+import hudson.model.AdministrativeMonitor;
+import hudson.model.ManagementLink;
+import java.util.Locale;
+
+/**
+ *  Definition of a badge that can be returned by a {@link ManagementLink} implementation.
+ *  The badge is shown as a small overlay over the corresponding icon on the {@code Manage Jenkins} page, can display additional information
+ *  in a tooltip and change it's color depending on the severity.
+ *
+ *  <p>
+ *  A badge mainly serves as a fast feedback for the corresponding management page.
+ *  It could be used to just display some short status information or hint that some action can be taken. E.g. the badge on {@code Manage Plugins}
+ *  shows information about the number of available updates and in its tooltip additionally how many updates contain incompatible changes or fix
+ *  security vulnerabilities. It also changes it's color when there are security fixes available.
+ *
+ *  <p>
+ *  A badge might display the same information as an {@link AdministrativeMonitor}. While an {@link AdministrativeMonitor}
+ *  can be disabled, a badge will be always shown. E.g. the badge of {@link OldDataMonitor.ManagementLinkImpl} always shows the number of old data entries.
+ *
+ *  @since TODO
+ */
+public class Badge {
+
+    private final String text;
+    private final String tooltip;
+    private final Severity severity;
+
+    /**
+     * Create a new Badge
+     *
+     * @param text  The text to be shown in the badge.
+     *      Keep it short, ideally just a number. More than 6 or 7 characters do not look good. Do not put in blanks
```

Comment:
what do you mean by blanks?

Reply:
A whitespace can lead to a line break getting rendered. See the first outdated conversation where I added some screenshots

Reply:
what about?

```suggestion
     *      Keep it short, ideally just a number. More than 6 or 7 characters do not look good. Avoid spaces as they will lead to line breaks.
```